### PR TITLE
feat(egh): set up instructor invites table and dashboard

### PR DESCRIPTION
**Pull request by zacjones93**
This enables working on the instructor invites project [EGG-389](https://linear.app/skillrecordings/issue/EGG-389/invite-instructors-through-course-builder) by setting up an invites table that will track the state of an invite as it's sent out and accepted by instructor. The invites table is set up to be local to egghead course builder only under `db/schemas`

This also sets up UI to send those invites out for admin:
image

## TODO
- [ ] open deploy request for egghead database

gif

### CODE REVIEW

File: apps/egghead/src/app/admin/_components/instructor-data-table.tsx [hunk 1]
```diff
@@ -107,12 +107,6 @@ const InstructorDataTable: React.FC<{ users: UserRow[] }> = ({ users }) => {
 						className="w-full max-w-sm pl-8"
 					/>
 				</div>
-				<Button
-					className="w-full disabled:cursor-not-allowed disabled:opacity-40 sm:w-auto"
-					disabled
-				>
-					<Plus className="mr-2 h-4 w-4" /> Add New Instructor
-				</Button>
```
Comment (zacjones93) [inline apps/egghead/src/app/admin/_components/instructor-data-table.tsx:115]: this button wasn't doing anything and with invites in a different tab didn't make sense on the instructors admin panel

File: apps/egghead/src/app/admin/instructors/invite/page.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+import { InviteInstructorForm } from './_components/invite-instructor-form'
+import InvitesDataTable from './_components/invites-data-table'
+
+export default async function InstructorsPage() {
+	const { ability } = await getServerAuthSession()
+	if (!ability.can('manage', 'all')) {
+		notFound()
+	}
+	const instructorInvites = await db.select().from(invites)
```
Comment (zacjones93) [inline apps/egghead/src/app/admin/instructors/invite/page.tsx:15]: might need to set up a invites lib file but this is the only page for now pulling this data in

File: apps/egghead/src/db/schema.ts [hunk 3]
```diff
@@ -1,7 +1,14 @@
 import { mysqlTable } from '@/db/mysql-table'
+import {
+	getInvitesRelationsSchema,
+	getInvitesSchema,
+} from '@/db/schemas/invites'
 
 import { getCourseBuilderSchema } from '@coursebuilder/adapter-drizzle/mysql'
 
+export const invites = getInvitesSchema(mysqlTable)
+export const invitesRelations = getInvitesRelationsSchema(mysqlTable)
```
Comment (zacjones93) [inline apps/egghead/src/db/schema.ts:10]: extending the egghead course builder database schema was surprisingly easy

File: apps/egghead/src/db/schemas/invites.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+		userId: varchar('userId', { length: 255 }),
+		createdAt: timestamp('createdAt').defaultNow(),
+		expiresAt: timestamp('expiresAt').default(
+			sql`(CURRENT_TIMESTAMP + INTERVAL 7 DAY)`,
+		),
+		canceledAt: timestamp('canceledAt'),
+		confirmedAt: timestamp('confirmedAt'),
+		completedAt: timestamp('completedAt'),
+	})
+}
```
Comment (zacjones93) [inline apps/egghead/src/db/schemas/invites.ts:56]: This table largely follow purchase transfer structure with slightly simplified states

File: apps/egghead/src/inngest/functions/instructor-invite-created.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+		// 	})
+		// })
+
+		return {
+			// sendResponse,
+			// email,
+			email: event.data.email,
+		}
+	},
+)
```
Comment (zacjones93) [inline apps/egghead/src/inngest/functions/instructor-invite-created.ts:55]: stubbed out the inngest workflow that we'll use to coordinate instructor invite. Currently it only adds an invite to the invite table

**Review by zacjones93 — COMMENTED** [review]
This is step 1 for instructor invites.

Next step will be to send an email to the `inviteEmail` to accept the invite and create instructor account (the core flow)

After that we will set up nice-to-haves like creating a slack channel and inviting everyone that should be there to it.

A stretch goal would be to allow instructor to input an email of their choosing instead of accepting invite to the current email

File: apps/egghead/src/db/schema.ts (hunk 3 above)
Comment (joelhooks) [reply apps/egghead/src/db/schema.ts:10]: like that we kept this "local"

**Review by joelhooks — APPROVED** [review]

**Pull request merged**